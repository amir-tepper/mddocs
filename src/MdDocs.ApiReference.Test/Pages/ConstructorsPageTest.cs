using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Grynwald.MdDocs.ApiReference.Configuration;
using Grynwald.MdDocs.ApiReference.Model;
using Grynwald.MdDocs.ApiReference.Model.XmlDocs;
using Grynwald.MdDocs.ApiReference.Pages;
using Microsoft.Extensions.Logging.Abstractions;

namespace Grynwald.MdDocs.ApiReference.Test.Pages
{
    /// <summary>
    /// Tests for <see cref="ConstructorsPage" />
    /// </summary>
    public class ConstructorsPageTest : PageTestBase<ConstructorDocumentation, ConstructorsPage>
    {
        protected override ConstructorsPage CreatePage(ConstructorDocumentation model, ApiReferenceConfiguration configuration)
        {
            return new ConstructorsPage(NullLinkProvider.Instance, configuration, model, NullLogger.Instance);
        }

        protected override ConstructorDocumentation CreateSampleModel()
        {
            var assembly = Compile(@"
                namespace MyNamespace
                {
                    public class Class1
                    {
                        public Class1()
                        { }
                    }
                }
            ");

            var assemblyDocumentation = new AssemblyDocumentation(assembly, NullXmlDocsProvider.Instance, NullLogger.Instance);
            return assemblyDocumentation.MainModuleDocumentation.Types.Single().Constructors!;
        }
    }
}
