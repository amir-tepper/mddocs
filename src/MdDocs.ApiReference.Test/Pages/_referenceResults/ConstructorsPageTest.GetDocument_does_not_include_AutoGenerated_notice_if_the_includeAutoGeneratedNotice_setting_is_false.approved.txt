# Class1 Constructors

**Declaring Type:** Class1  
**Namespace:** MyNamespace  
**Assembly:** DynamicTestAssembly  
**Assembly Version:** 0.0.0.0

```csharp
public Class1();
```
___

*Documentation generated by [MdDocs](https://github.com/ap0llo/mddocs)*
